src/ia-dropdown.ts

```diff
@@ -22,6 +22,26 @@ export class IaDropdown extends LitElement {
 
   @property({ type: Boolean, attribute: true }) displayCaret = false;
 
+  /**
+   * To allow customizing whether the dropdown menu automatically closes
+   * when an option is selected.
+   */
+  @property({ type: Boolean, attribute: true }) closeOnSelect = false;
+
+  /**
+   * To allow customizable behavior where clicking on the caret or
+   * the main button itself should be handled differently.
+   */
+  @property({ type: Boolean, attribute: true }) openViaButton = true;
```

Comment:
this does not happen currently, can we default to false?

Reply:
I think the current behavior is to open the dropdown when either the main button or its caret are clicked (no distinction). So for that reason I defaulted both options to `true` so that consumers will get the same behavior without any changes.

Reply:
Ah, perhaps the comment on this is confusing the property's purpose. I'll rewrite that for clarity.

Reply:
New comment should hopefully make the purpose clearer (and why it's default `true`)